open-source-services.md

```diff
@@ -28,19 +28,47 @@ Thus we must address common questions about the requirements:
+
+For an Open Source code base to run as an Open Source service, there are a number of [Open Works](https://fossrit.github.io/open-work-definition/) around it needed to make it run in a cloud or service environment: scripts, documentation, policies, configurations, and so forth.
+
+### Q: What "assets and dependencies" should be Open Source?
 
 **A:** Make all parts of the service open by default, except where law, security, privacy, or common sense says otherwise.
 A general rule of thumb is if you commit it to a repository, it should be open.
 Assets that should not be shared are ones where you don’t have redistribution rights (such as a RHEL VM image, or private information, and personal identifiable information), or things that are actually secret: Don’t share keys, passwords, or tokens.
 
 If changing configuration is an activity that a typical team member does in order to change the service, then you should share that configuration publicly, in order to meet **requirement #2**.
 
-It is _not requried_ that all tooling dependencies and service dependencies used during the building, testing, deployment and operating of the service are themselves Open Source.
+It is _not required_ that all tooling dependencies and service dependencies used during the building, testing, deployment, and operating of the service are themselves Open Source.
+
+It is _recommended_ that you include sufficient configuration to operate the service, or some base default version of configuration.
+Putting secrets in your configuration is a bad practice that should be avoided, regardless of Open Source.
+
+### Q: What "assets and dependencies" MUST be Open too?
 
-It is _recommended_ that you include sufficient configuration to operate the service, or some base default version of configuration. Putting secrets in your configuration is a bad practice that should be avoided, regardless of Open Source.
+**A:** It is _not required_ that all dependencies of a service are Open Source, or have Open Source alternatives.
+For example, a service may interact with a proprietary authentication system, or run on a proprietary platform.
+
+The requirements together serve to enable others to change the service code so that its proprietary dependencies can be replaced with Open Source ones, should that be desired.
+
+The best approach to take depends on if you are working with a new service or an existing service:
+1. For a new service, the "Open by default" practice means you limit your use of private code or non-Open services, platforms, etc.
+If you do have to choose a private/non-Open solution, make it a deliberate choice, such as when it is required in the environment where you build or run the service.
+2. For an existing service, you need to do a gap analysis to discover what assets and dependencies are private/non-Open.
+Then determine for each asset or dependency if it is necessary to the service being able to run in another environment
```

Comment:
Why are these latter steps necessary? Running in another environment, while desirable attribute of many services, is not one of these requirements.